src/Tickets/Emails/Email_Abstract.php

```diff
@@ -391,6 +381,260 @@ public function get_edit_url() {
+		 *
+		 * @param string $recipient  The email recipient.
+		 * @param string $id         The email id.
+		 * @param string $template   Template name.
+		 */
+		$recipient = apply_filters( 'tec_tickets_emails_recipient', $recipient, $this->id, $this->template );
+
+		/**
+		 * Allow filtering the email recipient for Completed Order.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $recipient  The email recipient.
+		 * @param string $id         The email id.
+		 * @param string $template   Template name.
+		 */
+		$template_name = $this->template;
+		$recipient = apply_filters( "tec_tickets_emails_{$template_name}_recipient", $recipient, $this->id, $this->template );
+
+		return $this->format_string( $recipient );
+	}
+
+	/**
+	 * Get the subject of the email.
+	 * 
+	 * @since TBD
+	 * 
+	 * @return string
+	 */
+	public function get_subject(): string {
+		$option_key = $this->get_option_key( 'subject' );
+		$subject = tribe_get_option( $option_key, $this->get_default_subject() );
+
+		// @todo: Probably we want more data parsed, or maybe move the filters somewhere else as we're always gonna
+
+		/**
+		 * Allow filtering the email subject globally.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $subject  The email subject.
+		 * @param string $id       The email id.
+		 * @param string $template Template name.
+		 */
+		$subject = apply_filters( 'tec_tickets_emails_subject', $subject, $this->id, $this->template );
+
+		/**
+		 * Allow filtering the email subject.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $subject  The email subject.
+		 * @param string $id       The email id.
+		 * @param string $template Template name.
+		 */
+		$template_name = $this->template;
+		$subject = apply_filters( "tec_tickets_emails_{$template_name}_subject", $subject, $this->id, $this->template );
+
+		return $this->format_string( $subject );
+	}
+
+	/**
+	 * Get email heading.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The email heading.
+	 */
+	public function get_heading(): string {
+		$option_key = $this->get_option_key( 'heading' );
+		$heading = tribe_get_option( $option_key, $this->get_default_heading() );
+
+		// @todo: Probably we want more data parsed, or maybe move the filters somewhere else as we're always gonna
+
+		/**
+		 * Allow filtering the email heading globally.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $heading  The email heading.
+		 * @param string $id       The email id.
+		 * @param string $template Template name.
+		 */
+		$heading = apply_filters( 'tec_tickets_emails_heading', $heading, $this->id, $this->template );
+
+		/**
+		 * Allow filtering the email heading for Completed Order.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $heading  The email heading.
+		 * @param string $id       The email id.
+		 * @param string $template Template name.
+		 */
+		$template_name = $this->template;
+		$heading = apply_filters( "tec_tickets_emails_{$template_name}_heading", $heading, $this->id, $this->template );
```

Comment:
The same comment about `$template_name`.

Reply:
We'll use the new `slug` variable.